Network/Socket/ByteString/IO.hsc

```diff
@@ -248,3 +266,87 @@ withIOVec cs f =
         unsafeUseAsCStringLen s $ \(sPtr, sLen) ->
         poke ptr $ IOVec sPtr (fromIntegral sLen)
 #endif
+
+-- | Send data from the socket using sendmsg(2).
+sendMsg :: Socket       -- ^ Socket
+        -> SockAddr     -- ^ Destination address
+        -> [ByteString] -- ^ Data to be sent
+        -> [Cmsg]       -- ^ Control messages
+        -> [MsgFlag]    -- ^ Message flags
```

Comment:
I would let the user specify whatever combination of flags he wants as an integral value. The user can apply appropriate bit operations to bitwise-or together the desired flags, the API just exports some pre-defined constants.  Thus no list of flags, just a `CInt`. However the newtype for the flag pattern synonyms can be a monoid under bitwise-or, and also provide direct conversions to/from `CInt`.